prusti-viper/src/encoder/procedure_encoder.rs

```diff
@@ -4780,15 +4889,14 @@ impl<'p, 'v: 'p, 'tcx: 'v> ProcedureEncoder<'p, 'v, 'tcx> {
             }
 
             &mir::AggregateKind::Closure(def_id, _substs) => {
-                //assert!(self.encoder.is_spec_closure(def_id), "closure: {:?}", def_id);
-                // Specification only. Just ignore in the encoding.
-                // FIXME: Filtering of specification blocks is broken, so we need to handle this here.
-              if self.encoder.is_spec_closure(def_id) {
-                // Specification only. Just ignore in the encoding.
-                // FIXME: Filtering of specification blocks is broken, so we need to handle this here.
-                Vec::new()
+                if self.encoder.is_spec_closure(def_id) {
+                    // Specification only. Just ignore in the encoding.
+                    // FIXME: Filtering of specification blocks is broken, so we need to handle this here.
```

Comment:
I guess this should be resolved since #176 ?

Reply:
This branch should be unreachable, because we don't encode to Viper the basic blocks used for specifications:

https://github.com/viperproject/prusti-dev/blob/d27d8fa49e08a058ac369be503825d4f283013b7/prusti-viper/src/encoder/procedure_encoder.rs#L900-L906

I would try replacing the `if` with a `debug_assert!(!self.encoder.is_spec_closure(def_id))`.